Sources/BigInt/BigInt.swift

```diff
@@ -330,37 +330,29 @@ extension BigInt: SignedNumeric {
 extension BigInt: BinaryInteger {
 
   public init?<T>(exactly source: T) where T: BinaryFloatingPoint {
-    if (source.isNaN || source.isInfinite) ||
-      (source.rounded(.towardZero) != source) {
+    guard source.isFinite, source == source.rounded(.towardZero) else {
       return nil
     }
-
     self.init(source)
   }
 
   public init<T>(_ source: T) where T: BinaryFloatingPoint {
-    precondition(
-      !(source.isNaN || source.isInfinite),
-      "\(type(of: source)) value cannot be converted to BigInt because it is either infinite or NaN"
-    )
-
-    let isNegative = source < 0.0
-    var float = isNegative ? -source : source
+    precondition(source.isFinite, "BigInt(\(source)) failed")
+    precondition(T.significandBitCount < .max)
 
-    if let _ = UInt(exactly: T.greatestFiniteMagnitude) {
-      words = [UInt(float)]
-    } else {
-      var words = Words()
-      let radix = T(sign: .plus, exponent: T.Exponent(UInt.bitWidth), significand: 1)
-      repeat {
-        let digit = UInt(float.truncatingRemainder(dividingBy: radix))
-        words.append(digit)
-        float = (float / radix).rounded(.towardZero)
-      } while float != 0
+    let source = source.rounded(.towardZero)
+    let isNegative = source < 0
 
-      self.words = words
+    guard !source.isZero else {
+      self = 0
```

Comment:
`words = [0]` is almost certainly going to be much faster

Reply:
Isn't that basically what your ExpressibleByIntegerLiteral conformance already does?

Another option is to implement and use `static var zero`, versus the [default implementation][].

[default implementation]: <https://github.com/apple/swift/blob/958326151c63ab3d8892dddd5b960125e3b97a37/stdlib/public/core/Integers.swift#L143-L152>

Reply:
Yes, kind of? It probably all comes to the same thing in optimized code, now that I think about it, so I'm OK with `self = 0` until proven otherwise.